src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1924,6 +1926,149 @@ def remove_comments(sql_query: str, uuids: List[str]) -> str:
     )
 
 
+class TimeTravelConfig(NamedTuple):
+    """Configuration for time travel operations."""
+
+    mode: str  # 'at' or 'before'
```

Comment:
nit: you can just rename this to "time_travel_mode" for consistency, such that you do not need to overwrite the `asdict_for_table`